x/lockup/keeper/lock.go

```diff
@@ -48,18 +48,31 @@ func (k Keeper) BeginUnlockAllNotUnlockings(ctx sdk.Context, account sdk.AccAddr
 }
 
 // AddToExistingLock adds the given coin to the existing lock with the same owner and duration.
-// Returns an empty array of period lock when a lock with the given condition does not exist.
-func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) ([]types.PeriodLock, error) {
+// Returns the updated lock ID if successfully added coin, returns 0 when a lock with
+// given condition does not exist.
+func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) (uint64, error) {
 	locks := k.GetAccountLockedDurationNotUnlockingOnly(ctx, owner, coin.Denom, duration)
-	// if existing lock with same duration and denom exists, just add there
-	if len(locks) > 0 {
-		lock := locks[0]
-		_, err := k.AddTokensToLockByID(ctx, lock.ID, owner, coin)
-		if err != nil {
-			return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
-		}
+
+	// if no lock exists for the given owner + denom + duration, return an error
+	if len(locks) < 1 {
+		return 0, sdkerrors.Wrap(types.ErrLockupNotFound, "lock with given condition does not exist")
+	}
+
+	// if existing lock with same duration and denom exists, add to the existing lock
+	// there should only be a single lock with the same duration + token, thus we take the first lock
+	lock := locks[0]
+	_, err := k.AddTokensToLockByID(ctx, lock.ID, owner, coin)
+	if err != nil {
+		return 0, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
 	}
-	return locks, nil
+
+	return lock.ID, nil
+}
+
+// HasLock returns true if lock with the given condition exists
+func (k Keeper) HasLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) bool {
```

Comment:
Can we add a table-driven test for this method as well, please? Also, it might make sense to have it unexported since we, currently, only call it from the same package.